blocks/extensions/monitoring.py

```diff
@@ -139,3 +140,17 @@ def do(self, callback_name, *args):
             self.add_records(self.main_loop.log,
                              self._buffer.get_aggregated_values().items())
             self._buffer.initialize_aggregators()
+
+
+class CrossEntropy(MonitoredQuantity):
+    def __init__(self, **kwargs):
+        super(CrossEntropy, self).__init__(**kwargs)
+        self.total_cross_entropy, self.examples_seen = 0.0, 0
+
```

Comment:
Right now you keep adding and adding and adding to `self.examples_seen` in each epoch. You should reset the counters to 0 at the start of each epoch.